### PR TITLE
Orchestrator otic tests

### PR DESCRIPTION
**Pull request by MaGaMeGa**

### CODE REVIEW

File: orchestrator/src/test/java/eu/arrowhead/core/orchestrator/matchmaking/RandomWeightedCloudMatchmakerTest.java [hunk 1]
```diff
@@ -0,0 +1,490 @@
...
+			
+			if (cloudResponseDTO.getName().equalsIgnoreCase(providerCloud1.getName())) {
+				
+				++provider1;
+			}else {
+				++provider2;
+			}
+		}
+		
+		Assert.assertTrue( provider1 > provider2 );
```
Comment (rbocsi) [inline orchestrator/src/test/java/eu/arrowhead/core/orchestrator/matchmaking/RandomWeightedCloudMatchmakerTest.java]: Most of the time this is true, but you can't guarantee it. There is a very little probability when provider2 is greater than provider1. I don't know how to handle this. Maybe you should just delete this test.

File: orchestrator/src/test/java/eu/arrowhead/core/orchestrator/matchmaking/RandomWeightedCloudMatchmakerTest.java [hunk 2]
```diff
@@ -0,0 +1,490 @@
...
+				unsuccessfulRequests);
+		
+		final CloudMatchmakingParameters params = new CloudMatchmakingParameters(
+				gsdQueryResultDTO, 
+				preferredClouds, 
+				onlyPreferred);	
+		
+		final CloudResponseDTO cloudResponseDTO = algorithm.doMatchmaking(params);
+		
+		Assert.assertTrue( cloudResponseDTO.getName() == null );
```
Comment (rbocsi) [inline orchestrator/src/test/java/eu/arrowhead/core/orchestrator/matchmaking/RandomWeightedCloudMatchmakerTest.java]: use assertNull instead

File: orchestrator/src/test/java/eu/arrowhead/core/orchestrator/matchmaking/RandomWeightedCloudMatchmakerTest.java [hunk 3]
```diff
@@ -0,0 +1,490 @@
...
+				unsuccessfulRequests);
+		
+		final CloudMatchmakingParameters params = new CloudMatchmakingParameters(
+				gsdQueryResultDTO, 
+				preferredClouds, 
+				onlyPreferred);	
+		
+		final CloudResponseDTO cloudResponseDTO = algorithm.doMatchmaking(params);
+		
+		Assert.assertTrue( cloudResponseDTO.getName() != null );
```
Comment (rbocsi) [inline orchestrator/src/test/java/eu/arrowhead/core/orchestrator/matchmaking/RandomWeightedCloudMatchmakerTest.java]: use assertNotNull instead

File: orchestrator/src/test/java/eu/arrowhead/core/orchestrator/matchmaking/RandomWeightedCloudMatchmakerTest.java [hunk 4]
```diff
@@ -0,0 +1,490 @@
...
+		final GSDQueryResultDTO gsdQueryResultDTO = null;
+		
+		final CloudMatchmakingParameters params = new CloudMatchmakingParameters(
+				gsdQueryResultDTO, 
+				preferredClouds, 
+				onlyPreferred);	
+		
+		final CloudResponseDTO cloudResponseDTO = algorithm.doMatchmaking(params);
+		
+		Assert.assertTrue( cloudResponseDTO.getName() == null );
```
Comment (rbocsi) [inline orchestrator/src/test/java/eu/arrowhead/core/orchestrator/matchmaking/RandomWeightedCloudMatchmakerTest.java]: use assertNull instead

File: orchestrator/src/test/java/eu/arrowhead/core/orchestrator/matchmaking/RandomWeightedCloudMatchmakerTest.java [hunk 5]
```diff
@@ -0,0 +1,490 @@
...
+				unsuccessfulRequests);
+		
+		final CloudMatchmakingParameters params = new CloudMatchmakingParameters(
+				gsdQueryResultDTO, 
+				preferredClouds, 
+				onlyPreferred);	
+		
+		final CloudResponseDTO cloudResponseDTO = algorithm.doMatchmaking(params);
+		
+		Assert.assertTrue( cloudResponseDTO.getName() == null );
```
Comment (rbocsi) [inline orchestrator/src/test/java/eu/arrowhead/core/orchestrator/matchmaking/RandomWeightedCloudMatchmakerTest.java]: use assertNull instead

File: orchestrator/src/test/java/eu/arrowhead/core/orchestrator/matchmaking/RandomWeightedCloudMatchmakerTest.java [hunk 6]
```diff
@@ -0,0 +1,490 @@
...
```
Comment (rbocsi) [inline orchestrator/src/test/java/eu/arrowhead/core/orchestrator/matchmaking/RandomWeightedCloudMatchmakerTest.java]: use assertNull instead

File: orchestrator/src/test/java/eu/arrowhead/core/orchestrator/matchmaking/RandomWeightedCloudMatchmakerTest.java [hunk 7]
```diff
@@ -0,0 +1,490 @@
...
+			
+			if (cloudResponseDTO.getName().equalsIgnoreCase(providerCloud1.getName())) {
+				
+				++provider1;
+			}else {
+				++provider2;
+			}
+		}
+		
+		Assert.assertTrue( provider1 > provider2 );
```
Comment (rbocsi) [inline orchestrator/src/test/java/eu/arrowhead/core/orchestrator/matchmaking/RandomWeightedCloudMatchmakerTest.java]: see the comments at the first test. You can't guarantee that result.

File: orchestrator/src/test/java/eu/arrowhead/core/orchestrator/service/OrchestratorDriverTest.java [hunk 8]
```diff
@@ -219,14 +219,50 @@ public void testQueryServiceRegistryBySystemIdOk() {
...
-		
 		final SystemResponseDTO responseDTO = new SystemResponseDTO();
 		
 		when(arrowheadContext.containsKey(any(String.class))).thenReturn(true);
 		when(arrowheadContext.get(any(String.class))).thenReturn(queryBySystemIdUri);
 		when(httpService.sendRequest(eq(queryBySystemIdUriForTestRequest), eq(HttpMethod.GET), eq(SystemResponseDTO.class))).thenReturn(new ResponseEntity<SystemResponseDTO>(responseDTO, HttpStatus.OK));
 		
+		final SystemResponseDTO systemResponseDTO = orchestratorDriver.queryServiceRegistryBySystemId( 1L );
+		
+		Assert.assertTrue( systemResponseDTO != null );
```
Comment (rbocsi) [inline orchestrator/src/test/java/eu/arrowhead/core/orchestrator/service/OrchestratorDriverTest.java]: use assertNotNull instead

File: orchestrator/src/test/java/eu/arrowhead/core/orchestrator/service/OrchestratorDriverTest.java [hunk 9]
```diff
@@ -261,6 +297,40 @@ public void testQueryServiceRegistryBySystemRequestDTOOk() {
 		when(arrowheadContext.get(any(String.class))).thenReturn(queryBySystemDTOUri);
 		when(httpService.sendRequest(eq(queryBySystemDTOUri), eq(HttpMethod.POST), eq(SystemResponseDTO.class), any(SystemRequestDTO.class))).thenReturn(new ResponseEntity<SystemResponseDTO>(responseDTO, HttpStatus.OK));
 		
+		final SystemResponseDTO systemResponseDTO = orchestratorDriver.queryServiceRegistryBySystemRequestDTO(requestDTO);
+		
+		Assert.assertTrue( systemResponseDTO != null );
```
Comment (rbocsi) [inline orchestrator/src/test/java/eu/arrowhead/core/orchestrator/service/OrchestratorDriverTest.java]: use assertNotNull instead

File: orchestrator/src/test/java/eu/arrowhead/core/orchestrator/service/OrchestratorDriverTest.java [hunk 10]
```diff
@@ -293,6 +363,40 @@ public void testDoGlobalServiceDiscoveryOk() {
 		when(arrowheadContext.get(any(String.class))).thenReturn(queryGSDUri);
 		when(httpService.sendRequest(eq(queryGSDUri), eq(HttpMethod.POST), eq(GSDQueryResultDTO.class), any(GSDQueryFormDTO.class))).thenReturn(new ResponseEntity<GSDQueryResultDTO>(responseDTO, HttpStatus.OK));
 		
+		final GSDQueryResultDTO gsdQueryResultDTO = orchestratorDriver.doGlobalServiceDiscovery(requestDTO);
+
+		Assert.assertTrue( gsdQueryResultDTO != null );
```
Comment (rbocsi) [inline orchestrator/src/test/java/eu/arrowhead/core/orchestrator/service/OrchestratorDriverTest.java]: use assertNotNull instead

File: orchestrator/src/test/java/eu/arrowhead/core/orchestrator/service/OrchestratorDriverTest.java [hunk 11]
```diff
@@ -325,9 +429,42 @@ public void testDoInterCloudNegotiationsOk() {
 		when(arrowheadContext.get(any(String.class))).thenReturn(queryICNUri);
 		when(httpService.sendRequest(eq(queryICNUri), eq(HttpMethod.POST), eq(ICNResultDTO.class), any(ICNRequestFormDTO.class))).thenReturn(new ResponseEntity<ICNResultDTO>(responseDTO, HttpStatus.OK));
 		
+		final ICNResultDTO icnResultDTO = orchestratorDriver.doInterCloudNegotiations(requestDTO);
+		
+		Assert.assertTrue( icnResultDTO != null );
```
Comment (rbocsi) [inline orchestrator/src/test/java/eu/arrowhead/core/orchestrator/service/OrchestratorDriverTest.java]: use assertNotNull instead

**Review by rbocsi — CHANGES_REQUESTED** [review]
see inline comments

File: orchestrator/src/test/java/eu/arrowhead/core/orchestrator/matchmaking/RandomWeightedCloudMatchmakerTest.java (hunk 1 above)
Comment (MaGaMeGa) [reply orchestrator/src/test/java/eu/arrowhead/core/orchestrator/matchmaking/RandomWeightedCloudMatchmakerTest.java]: deleted loop and renamed 
tests not null returned values

File: orchestrator/src/test/java/eu/arrowhead/core/orchestrator/matchmaking/RandomWeightedCloudMatchmakerTest.java (hunk 2 above)
Comment (MaGaMeGa) [reply orchestrator/src/test/java/eu/arrowhead/core/orchestrator/matchmaking/RandomWeightedCloudMatchmakerTest.java]: done

File: orchestrator/src/test/java/eu/arrowhead/core/orchestrator/matchmaking/RandomWeightedCloudMatchmakerTest.java (hunk 3 above)
Comment (MaGaMeGa) [reply orchestrator/src/test/java/eu/arrowhead/core/orchestrator/matchmaking/RandomWeightedCloudMatchmakerTest.java]: done

File: orchestrator/src/test/java/eu/arrowhead/core/orchestrator/matchmaking/RandomWeightedCloudMatchmakerTest.java (hunk 4 above)
Comment (MaGaMeGa) [reply orchestrator/src/test/java/eu/arrowhead/core/orchestrator/matchmaking/RandomWeightedCloudMatchmakerTest.java]: done

File: orchestrator/src/test/java/eu/arrowhead/core/orchestrator/matchmaking/RandomWeightedCloudMatchmakerTest.java (hunk 5 above)
Comment (MaGaMeGa) [reply orchestrator/src/test/java/eu/arrowhead/core/orchestrator/matchmaking/RandomWeightedCloudMatchmakerTest.java]: done

File: orchestrator/src/test/java/eu/arrowhead/core/orchestrator/matchmaking/RandomWeightedCloudMatchmakerTest.java (hunk 6 above)
Comment (MaGaMeGa) [reply orchestrator/src/test/java/eu/arrowhead/core/orchestrator/matchmaking/RandomWeightedCloudMatchmakerTest.java]: done

File: orchestrator/src/test/java/eu/arrowhead/core/orchestrator/matchmaking/RandomWeightedCloudMatchmakerTest.java (hunk 7 above)
Comment (MaGaMeGa) [reply orchestrator/src/test/java/eu/arrowhead/core/orchestrator/matchmaking/RandomWeightedCloudMatchmakerTest.java]: done

File: orchestrator/src/test/java/eu/arrowhead/core/orchestrator/service/OrchestratorDriverTest.java (hunk 8 above)
Comment (MaGaMeGa) [reply orchestrator/src/test/java/eu/arrowhead/core/orchestrator/service/OrchestratorDriverTest.java]: done

**Pull request merged**